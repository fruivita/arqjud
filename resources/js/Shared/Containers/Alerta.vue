<!--
    Container para as mensagens de alerta.
    Não se trata de notificação, mas de alerta exibido no corpo da página.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->
<template>
    <div class="rounded bg-red-300 p-3 font-mono text-red-900">
        <slot></slot>
    </div>
</template>
