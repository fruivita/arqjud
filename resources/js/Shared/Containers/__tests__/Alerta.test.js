/**
 * Testes para o componente Alerta.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Alerta from '@/Shared/Containers/Alerta.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(Alerta, { ...options });
    };
});

describe('Alerta', () => {
    // Caminho feliz
    test('renderiza o componente respeitando o snapshot', () => {
        expect(mountFunction().html()).toMatchSnapshot();
    });

    test('renderiza o componente com o slot respeitando o snapshot', () => {
        const wrapper = mountFunction({ slots: { default: 'foo' } });

        expect(wrapper.html()).toMatchSnapshot();
    });
});
