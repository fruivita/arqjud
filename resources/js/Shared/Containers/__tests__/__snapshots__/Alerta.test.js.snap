// Vitest Snapshot v1

exports[`Alerta > renderiza o componente com o slot respeitando o snapshot 1`] = `"<div class=\\"rounded bg-red-300 p-3 font-mono text-red-900\\">foo</div>"`;

exports[`Alerta > renderiza o componente respeitando o snapshot 1`] = `"<div class=\\"rounded bg-red-300 p-3 font-mono text-red-900\\"></div>"`;
